mlflow/langchain/langchain_tracer.py

```diff
@@ -266,6 +270,28 @@ def on_llm_start(
         mlflow_messages = [ChatMessage(role="user", content=prompt) for prompt in prompts]
         set_span_chat_messages(span, mlflow_messages)
 
+        if tools := self._extract_tool_definitions(kwargs):
+            set_span_chat_tools(span, tools)
+
+    def _extract_tool_definitions(self, kwargs: dict[str, Any]) -> list[ChatTool]:
+        raw_tools = kwargs.get("invocation_params", {}).get("tools", [])
+        tools = []
+        for raw_tool in raw_tools:
+            # First, try to parse the raw tool dictionary as OpenAI-style tool
+            try:
+                tool = ChatTool.validate_compat(raw_tool)
+                tools.append(tool)
+            except pydantic.ValidationError:
+                # If not OpenAI style, just try to extract the name and descriptions.
+                tool = ChatTool(
+                    type="function",
+                    function=FunctionToolDefinition(
+                        name=raw_tool.get("name"), description=raw_tool.get("description")
```

Comment:
Since name is a required field in FunctionToolDefinition, probably we want to make sure "name" exists in raw_tool here?

Reply:
Ah that makes sense. Well it is probably hard to make sure that as the tool definition is fully up to the LLM provider; but I will update the logic to only log attribute when we can find the name field, otherwise log a message saying "failed to capture tool definition blah blah".